core/state/statedb.go

```diff
@@ -585,6 +591,12 @@ func (s *StateDB) clearJournalAndRefund() {
 func (s *StateDB) Commit(deleteEmptyObjects bool) (root common.Hash, err error) {
 	defer s.clearJournalAndRefund()
 
+	dirtyObjects := s.journal.flatten()
+
+	for addr, v := range dirtyObjects {
```

Comment:
Nitpick, you can replace `dirtyObjects` with `s.journal.flatten()`. Less code, cleaner.